windows-gaming-driver/src/controller.rs

```diff
@@ -58,23 +64,27 @@ impl Controller {
 
     pub fn ga_ping(&mut self) -> bool {
         // the idea is that someone else (timer) calls this periodically
-        assert!(self.ga_up);
-
-        if self.ga_pong_expected {
-            // the last ping wasn't even answered
-            // we conclude that the ga has died
-            self.ga_up = false;
-            self.set_io_attached(false, true);
-            false
-        } else {
-            self.ga_pong_expected = true;
-            self.write_ga(GaCmd::Ping);
-            true
+        match self.ga {
+            State::Pinging => {
+                // the last ping wasn't even answered
+                // we conclude that the ga has died
+                self.ga = State::Down;
+                self.set_io_attached(false, true);
+                false
+            }
+            State::Up => {
+                self.ga = State::Pinging;
+                self.write_ga(GaCmd::Ping);
+                true
+            }
+            _ => false,
         }
     }
 
     pub fn ga_pong(&mut self) {
-        self.ga_pong_expected = false;
+        if self.ga == State::Pinging {
```

Comment:
We generally don't want to let the guest crash the driver. In this case, there is a straightforward 100% correct way to deal with the error case: Just ignore it. So that's what we do.